lib/internal/per_context/domexception.js

```diff
@@ -49,14 +49,23 @@ const disusedNamesSet = new SafeSet()
   .add('ValidationError');
 
 class DOMException {
-  constructor(message = '', name = 'Error') {
+  constructor(message = '', name = 'Error', cause = undefined) {
     ErrorCaptureStackTrace(this);
     internalsMap.set(this, {
       message: `${message}`,
-      name: `${name}`
+      name: `${name}`,
+      cause
     });
   }
 
+  get cause() {
```

Comment:
Per the debate (https://github.com/whatwg/webidl/pull/1179) that is happening in the PR that is updating `DOMException` to have a `cause` there is a disagreement on whether `cause` should be an own property or an accessor. For all intrinsic `Error`s, TC-39 decided that `cause` should be an own data property in order to distinguish between the case where `cause` is explicitly not given vs. `cause` is explicitly `undefined`

```js
const err1 = new Error('no cause');
console.log('cause' in err1); // false
console.log(err1.cause); // undefined

const err2 = new Error('undefined cause', { cause: undefined });
console.log('cause' in err2); // true
console.log(err2.cause); // undefined
```

From all appearances, it seems a decision is going to be made that `DOMException` should be different that intrinsic errors and will use an accessor instead, making it impossible to distinguish between these cases.

```js
const err1 = new DOMException('no cause', 'ABORT_ERR');
console.log('cause' in err1); // true
console.log(err1.cause); // undefined

const err2 = new DOMException('undefined cause', { name: 'ABORT_ERR', cause: undefined });
console.log('cause' in err2); // true
console.log(err2.cause); // undefined
```

The reasons for diverging do not seem that well defined beyond consistency with the way `name` and `message` are currently handled for `DOMException` but it's not really a compelling reason for me.

For our implementation, I'd rather we choose to align with TC-39's definition and intentionally diverge from the official definition that is likely to land in the WebIDL spec.

tl;dr ... let's not use an accessor for `cause`. Let's make it an own data property on the `DOMException` instance.

Reply:
https://github.com/nodejs/node/pull/44224#issuecomment-1243419207